util/dnsname/dnsname.go

```diff
@@ -65,6 +65,24 @@ func ToFQDN(s string) (FQDN, error) {
 	return FQDN(raw), nil
 }
 
+// ToFQDNAllowWildcard converts s to an FQDN, allowing a leading "*." for
+// wildcard DNS records. For example, "*.example.com" is valid and returns
+// "*.example.com.".
+func ToFQDNAllowWildcard(s string) (FQDN, error) {
+	if strings.HasPrefix(s, "*.") {
+		rest := s[2:]
```

Comment:
This could be
```suggestion
	if rest, ok := strings.CutPrefix(s, "*."); ok {
```

if you like.